
fix kernel crash/system hang when disabling monitor interface:
   echo 0 > /proc/brcm_monitor0
tested on: kernel 3.14

diff -p -r -u5 broadcom-sta_6.30.223.248-2_deb/src/wl/sys/wl_cfg80211_hybrid.c broadcom-sta_6.30.223.248-2_fix/src/wl/sys/wl_cfg80211_hybrid.c
--- broadcom-sta_6.30.223.248-2_deb/src/wl/sys/wl_cfg80211_hybrid.c
+++ broadcom-sta_6.30.223.248-2_fix/src/wl/sys/wl_cfg80211_hybrid.c
@@ -2631,11 +2631,19 @@ cfg80211_attach_out:
 	return err;
 }
 
 void wl_cfg80211_detach(struct net_device *ndev)
 {
-	struct wl_cfg80211_priv *wl = ndev_to_wl(ndev);
+	struct wl_cfg80211_priv *wl;
+	struct wireless_dev *wdev;
+	
+	wdev = ndev->ieee80211_ptr;
+	if (wdev == NULL) {
+                printk(KERN_ERR "[%s()] in ndev=%p: IEEE80211ptr=%p\n", __FUNCTION__, ndev, wdev);
+                return;
+        }
+	wl = ndev_to_wl(ndev);
 
 	wl_deinit_cfg80211_priv(wl);
 	wl_free_wdev(wl);
 }
 
